File: tests/fixtures/php/inheritance.php
<?php

class Animal {
    public string $name;
}

interface Swimmable {
    public function swim(): void;
}

interface Flyable {
    public function fly(): void;
}

trait HasLegs {
    public function walk(): void {}
}

class Duck extends Animal implements Swimmable, Flyable {
    use HasLegs;

    public function swim(): void {}
    public function fly(): void {}
    public function quack(): void {}
}

File: tests/fixtures/kotlin/inheritance.kt
package com.example.inheritance

open class Animal(val name: String) {
    open fun speak(): String = "..."
}

class Dog(name: String) : Animal(name) {
    override fun speak(): String = "Woof!"
}

interface Swimmable {
    fun swim(): String
}

interface Flyable {
    fun fly(): String
}

class Duck(name: String) : Animal(name), Swimmable, Flyable {
    override fun speak(): String = "Quack!"
    override fun swim(): String = "Swimming"
    override fun fly(): String = "Flying"
}
